frontend/README.md

```diff
@@ -79,21 +79,64 @@ kubectl port-forward svc/metadata-envoy-service 9090:9090
 
 ### Proxy to a real cluster
 
-This requires you already have a real KFP cluster, you can proxy requests to it.
-
-1. [Install Kubeflow Pipelines](https://www.kubeflow.org/docs/pipelines/installation/overview/) based on your use case and environment.
-1. Configure `kubectl` with access to your KFP cluster. (For GCP, follow [Access to GCP cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl) guide).
-1. Use the following table to determine which script to run.
+You can proxy requests from the UI running on your host machine to an actual KFP
+deployment, running on a remote or local Kubernetes cluster. This dramatically
+improves iteration time, especially since the docker build can take 20+ minutes.
+
+KFP can be deployed in single-user or multi-user mode. Since there's a delta in
+logic between between the two modes, automated tests and manual validation
+against a single-user cluster can still fail when deployed to a multi-user
+cluster. Validating against a multi-user cluster at some point in your
+development cycle is advisable, even though it requires jumping through a few
+extra hoops.
+
+#### Single-user
+
+1. Follow the [single-user KFP installation
+   instructions](https://www.kubeflow.org/docs/components/pipelines/operator-guides/installation/).
+2. Configure `kubectl` with access to your KFP cluster. (For GCP, follow [Access
+   to GCP
+   cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl)
+   guide).
+3. Use the following table to determine which script to run.
 
 | What to develop?                     | Script to run                                               | Extra notes                                                        |
 | ------------------------------------ | ----------------------------------------------------------- | ------------------------------------------------------------------ |
 | Client UI                            | `NAMESPACE=kubeflow npm run start:proxy`                    |                                                                    |
 | Client UI + Node server              | `NAMESPACE=kubeflow npm run start:proxy-and-server`         | You need to rerun the script every time you edit node server code. |
 | Client UI + Node server (debug mode) | `NAMESPACE=kubeflow npm run start:proxy-and-server-inspect` | Same as above, and you can use chrome to debug the server.         |
 
+#### Multi-user
+
+1. Install Kubernetes and deploy KFP to it on your your local machine by
+   following the [multi-user Kubeflow installation
+   instructions](https://github.com/kubeflow/manifests?tab=readme-ov-file#installation).
+2. Comment out
```

Comment:
Just a random thought, would steps 2 and 3 fail if they are future updates to index.tsx that change the lines? Should we instead paste both lines and instruct users to search for and replace specific lines.

Reply:
I share your concern about that. The only problem with copy-pasting the lines themselves is that they might change, in which case we're dealing with desync between documentation and code anyway.

I'm actively working on leveraging https://www.npmjs.com/package/react-inject-env to streamline this with env vars so that end users don't have to rebuild their application each time. I'll update these docs if / when that's merged.